Sources/Tracking/CustomerIOImplementation.swift

```diff
@@ -0,0 +1,110 @@
+import Foundation
+
+/**
+ Welcome to the Customer.io iOS SDK!
+
+ This class is where you begin to use the SDK.
+ You must have an instance of `CustomerIO` to use the features of the SDK.
+
+ To get an instance, you have 2 options:
+ 1. Use the already provided singleton shared instance: `CustomerIO.instance`.
+ This method is provided for convenience and is the easiest way to get started.
+
+ 2. Create your own instance: `CustomerIO(siteId: "XXX", apiKey: "XXX", region: Region.US)`
+ This method is recommended for code bases containing
+ automated tests, dependency injection, or sending data to multiple Workspaces.
+ */
+public class CustomerIOImplementation: CustomerIOInstance {
+    public var siteId: String? {
+        _siteId
+    }
+
+    private let _siteId: String
+
+    private let diGraph: DITracking
+
+    private let identifyRepository: IdentifyRepository
+
+    /**
+     Constructor for singleton, only.
+
+     Try loading the credentials previously saved for the singleton instance.
+     */
+    internal init(siteId: String) {
+        self._siteId = siteId
+
+        self.diGraph = DITracking.getInstance(siteId: siteId)
+
+        self.identifyRepository = diGraph.identifyRepository
```

Comment:
is there still value in having the concept of the `identifyRepository` - since we can get access to the `profileStore` directly? feels like it would simplify development by removing one more layer (similar to how the `MessagingPush` module just does the http calls directly within its methods)

Reply:
Good question. 

The idea behind a repository is re-usability and keeping this class more simple by delegating tasks. However, as you have stated, the logic is not re-used anywhere at the moment but this 1 place. 

I imagine that when a background queue comes, all of the logic can be moved out of this class and in another class that runs background queue operations. Let's consider removing repositories then?